### ISSUE TITLE
Pitest: add new mutator CONSTRUCTOR_CALLS 

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#6289]
This is **an unstable mutator**.

http://pitest.org/quickstart/mutators/#CONSTRUCTOR_CALLS

> The constructor call mutator replaces constructor calls with null values.

This is all they state on it being unstable:
> Please note that this mutation is fairly unstable and likely to cause NullPointerExceptions even with weak test suites.

Report with only new mutator added, and others disabled: http://rveach.no-ip.org/checkstyle/regression/pitest-reports/27/

I have seen mostly valid cases where we aren't verifying exception message, or using the initial value of the field. I have resolved 99% of the cases.
For throwing exceptions, the act of throwing `null` instead creates and throws a `NullPointerException`.

**Comment by rnveach** [comment]
@romani Some of my PRs are addressing the existing mutations without directly targeting them. See #6285 and #6268 .
I made sure I could finish this issue before opening it, so no new mutations will be left.

I have pretty much queued up all my work on the existing issues I have created, so I can go and tackle the existing mutations that are left. The set of mutators I have not created issues for (NON_VOID_METHOD_CALLS  and REMOVE_CONDITIONALS) have a bunch of survivors that I don't want to get into right now.
